File: python/core/bizzfuzzlinio.py
class BizzFuzzLinio:
	def __init__(self):
		self.messages = ["IT", "Linio", "Linianos"]

	def run(self):
		is_divisible_by_three = 0
		is_divisible_by_both = 0

		for i in range(1,101):
			is_divisible_by_both = i%3==0 and i%5==0
			is_divisible_by_three = i%3==0 and i%5>0

			if i%3==0 or i%5==0:
				print self.messages[(is_divisible_by_both*2)+is_divisible_by_three]
				continue

			print i

	def get_message_by_number(self, n):
		if n%3==0 or n%5==0:
			return self.messages[((n%3==0 and n%5==0)*2)+(n%3==0 and n%5>0)]

		return n

bizzfuzz = BizzFuzzLinio()

File: js/core/bizzfuzz-linio.js
var BizzFuzzLinio = /** @class */ (function () {
    function BizzFuzzLinio() {
        this.messages = ["IT", "Linio", "Linianos"];
    }
    /**
     * [run executes bizz fuzz linio algorithm]
     */
    BizzFuzzLinio.prototype.run = function () {
        var isDivisibleByBoth = 0;
        var isDivisibleByThree = 0;
        for (var i = 1; i <= 100; i++) {
            isDivisibleByBoth = Number(i % 3 == 0 && i % 5 == 0);
            isDivisibleByThree = Number(i % 3 == 0 && i % 5 > 0);
            if (i % 3 == 0 || i % 5 == 0) {
                console.log(this.messages[(isDivisibleByBoth * 2) + isDivisibleByThree]);
                continue;
            }
            console.log(i);
        }
    };
    /**
     * [getMessageByNumber given one number gets the message]
     */
    BizzFuzzLinio.prototype.getMessageByNumber = function (n) {
        return (n % 3 == 0 || n % 5 == 0) ? this.messages[(Number(n % 3 == 0 && n % 5 == 0) * 2) + Number(n % 3 == 0 && n % 5 > 0)] : String(n);
    };
    return BizzFuzzLinio;
}());

module.exports = new BizzFuzzLinio();